js/ui/windowManager.js

```diff
@@ -1379,6 +1382,67 @@ const WindowManager = new Lang.Class({
                                onOverwriteParams: [shellwm, actor]
                              });
         }
+        this._endlessCodingAddLauncher(actor);
+    },
+
+    _endlessCodingAddLauncher : function(actor) {
+        let window = actor.meta_window;
+        if (window.get_flatpak_id() !== 'org.gnome.gedit')
+            return;
+
+        let button = new St.Button({ style_class: 'view-source' });
+        let rect = window.get_frame_rect();
+        button.set_position(rect.x + rect.width - 100, rect.y + rect.height - 100);
+        Main.layoutManager.addChrome(button);
+        let positionChangedId = window.connect('position-changed', Lang.bind(this, function(window) {
+            let currentSession = this._currentEndlessCodingSessions.filter(function(session) {
+                return session.windowApp === window;
+            });
+            if (currentSession.length === 0)
+                return;
+            let rect = currentSession[0].windowApp.get_frame_rect();
+            currentSession[0].button.set_position(rect.x + rect.width - 100, rect.y + rect.height - 100);
+        }));
+        let windowsRestackedId = Main.overview.connect('windows-restacked', Lang.bind(this, this._endlessCodingWindowsRestacked, window));
+        button.connect ('clicked', Lang.bind(this, function(actor, event) {
+            Util.trySpawn(['flatpak', 'run', 'org.gnome.Builder']);
```

Comment:
From the docs:
```
// Runs @argv in the background. If launching @argv fails,  
// this will throw an error. 
```

We could use Utils.spawn(argv, errorNotifier) as well and take care ourselves of logging the error. Preferences?